source/test/TestApp/PlayFabEventTest.h

```diff
@@ -57,11 +64,19 @@ namespace PlayFabUnit
             static int eventFailCount;
             static std::string eventFailLog;
 
+            static const unsigned int numThreads = 6;
+            static const unsigned int numEventsPerThread = 5;
+            static const unsigned int numTotalThreadedEvents = numThreads * numEventsPerThread;
+            static std::vector<std::thread> testThreadPool;
+            static unsigned int numEventsHeard;
```

Comment:
as pointed out in other comments I have some concerns about the use of shared static state.

I might recommend you do a larger sweep of your test code to eliminate as much shared static state as possible. The only things that I think should even be considered candidates for shared static state should be values that have to persist between tests (such as eventPassCount and eventFailCount) but there's likely a way to rework those to not be static as well.

Reply:
That fuller sounding refactor sounds out of the scope of this particular pull request.

---
In reply to: [345444246](https://github.com/PlayFab/XPlatCoreTemplate/pull/44#discussion_r345444246) [](ancestors = 345444246)

Reply:
I agree that updating this file so that there is no shared static state is out of scope for this refactor, but I think even without doing that I don't think this thread pool needs to be static right?

---
In reply to: [345512283](https://github.com/PlayFab/XPlatCoreTemplate/pull/44#discussion_r345512283) [](ancestors = 345512283,345444246)